# Data transformation

This jupyter notebook contains open source code for transforming a data from the Australian Bureau of Statistics (ABS). The 2007 National Survey of Mental Health and Wellbeing (SMHWB)” collection [1] by the Australian Bureau of Statistics (ABS). This data collection contains information about Australians’ mental health and well-being in 2007. The ABS (https://www.abs.gov.au) is a national statistical agency in Australia, and provides a wide range of statistical data collections on economic, population, environmental, and social issues.

## Import library

In [1]:
import pandas as pd

## Download data

We download "*National Survey of Mental Health and Wellbeing: Summary of Results (43260do001_2007.xls)*" from ABS (https://www.abs.gov.au/statistics/health/mental-health/national-survey-mental-health-and-wellbeing-summary-results/2007#data-download). The data includes information about the 2007 National Survey of Mental Health and Wellbeing (SMHWB) that was designed to provide lifetime prevalence estimates for mental disorders.

In [18]:
input_file = '43260do001_2007.xls'
output_file = 'generated_data/transformed_data.xlsx'
writer = pd.ExcelWriter(output_file, engine = 'xlsxwriter')

## Read data

In [19]:
df1 = pd.read_excel(input_file, sheet_name = 'Table_1', skiprows=4, header=0, skipfooter= 3)
df2 = pd.read_excel(input_file, sheet_name = 'Table_3', skiprows=4, header=0, skipfooter= 3)

## Extract and transform the data

### Table 1: LIFETIME MENTAL DISORDERS

In [20]:
extract_columns = ['Unnamed: 0', 'Males', 'Females', 'Persons', 'Unnamed: 3', 'Unnamed: 7', 'Unnamed: 11']
extract_rows = ['Any Anxiety disorder', 'Any Affective disorder', 'Any Substance Use disorder', 'Any ifetime mental disorder']
extracted_df = pd.DataFrame(columns=['Unnamed: 0'])

for index, row in df1.iterrows():
    if row['Unnamed: 0'] in extract_rows:
        extracted_df = extracted_df.append(row)
extracted_df = extracted_df.dropna(subset=['Unnamed: 0'], how='all') # drop rows where it contains meaningless values
extracted_df = extracted_df[extract_columns]

extracted_df.rename(columns={"Unnamed: 0": "Disorder types", 
                             "Unnamed: 3": "Male proportion(%)", 
                             "Unnamed: 7": "Female proportion(%)", 
                             "Unnamed: 11":"Person proportion(%)"}, inplace=True) #rename columns
# save the data as excel file
extracted_df.to_excel(writer, sheet_name = 'lifetime')
extracted_df

Disorder types   Males  Females  Persons  Male proportion(%)  \
10         Any Anxiety disorder  1624.2   2580.8   4205.0                20.4   
15       Any Affective disorder   972.1   1433.3   2405.3                12.2   
20   Any Substance Use disorder  2816.7   1143.5   3960.3                35.4   
21  Any ifetime mental disorder  3822.0   3464.6   7286.6                48.1   

    Female proportion(%)  Person proportion(%)  
10                  32.0                  26.3  
15                  17.8                  15.0  
20                  14.2                  24.7  
21                  43.0                  45.5

We extract columns and rows from the "Table_1" in the *National Survey of Mental Health and Wellbeing: Summary of Results 2007* data file. Table_1 contains proportion of lifetime mental disorders by gender and disorder types. 
According to the transformed data, 45.5% of Australian reported experiencing a mental ill-health (disorder) at some point in their lifetime where male proportion size is around 5 percents greater than female's. 

### Table 3: 12-MONTHS MENTAL DISORDERS BY AGE GROUP

In [21]:
extract_columns = ['Unnamed: 0', 'Unnamed: 15'] 
extract_rows = ['PERSONS']

extracted_df2 = pd.DataFrame(columns=['Unnamed: 0'])

rows_no = 0
for index, row in df2.iterrows():
    if rows_no > 0:
        extracted_df2 = extracted_df2.append(row)
    if row['Unnamed: 0'] in extract_rows:
        rows_no = 7
    rows_no -= 1
    
extracted_df2 = extracted_df2.dropna(subset=['Unnamed: 0'], how='all') # drop rows where it contains meaningless values
extracted_df2 = extracted_df2[extract_columns]

extracted_df2.rename(columns={"Unnamed: 0":"Person age groups",
                             "Unnamed: 15":"Person proportion(%)"}, inplace=True) #rename columns
# save the data as excel file
extracted_df2.to_excel(writer, sheet_name = '12-months')
extracted_df2

Person age groups  Person proportion(%)
21             16–24                  26.4
22             25–34                  24.8
23             35–44                  23.3
24             45–54                  21.5
25             55–64                  13.6
26             65–74                   8.6

Table 3 includes information about 12-month mental disorders by age group. We extract and transform the data; the result showed in the above table. The proportion size getting decrease while age group are increased.

## Save the results

In [23]:
# save retuls
writer.save()
writer.close()

## Conclusion

In this Jupyter notebook, we demonstrated how to transform a data tables by using python code. The National Survey of Mental Health and Wellbeing: Summary of Results:2007 data from ABS has been transformed to excel file with two tables. First table contains proportion of lifetime mental disorders by gender and disorder types while second table contains proportion of 12-months mental disorders by age groups.  

# References

[1] Australian Bureau of Statistics, National Survey of Mental Health and Wellbeing: summary of results, Australian Bureau of Statistics, 2008. [Dataset] Available:https://www.abs.gov.au/statistics/health/mental-health/national-survey-mental-health-and-wellbeing-summary-results/2007. [Accessed: January 4, 2021].